### PR TITLE
Add models and examples reference

### LINKED ISSUE #1808: [Book] Add reference to models repo
We're starting to have more and more models available for our users so it would be nice to add a small paragraph in the book with a link to the models repo.

### LINKED ISSUE #1809: [Book] Add examples section
We currently have a basic workflow example in the guide which briefly mentions other examples available on the repo.

[As previously suggested](https://github.com/tracel-ai/burn/pull/1649/files/4da8c2ffdf80bf5b350912bd9e608530a60dae86#r1568791813), I think would be a great idea to have a separate section/page for the current examples available.

### PR DESCRIPTION
**Pull request by laggui**
Figured I'd tackle this before we eventually get to the release date and we still haven't done it.

### Checklist

- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

Closes #1808 
Closes #1809

### Changes

- Added reference to the models repo in the book
- Added the list of examples in the book and README

When rendered, the examples section looks like this:

image

### CODE REVIEW

File: burn-book/src/examples.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+The examples are unique files under the `examples` directory. Each file produces an executable file
+with the same name. Each example can then be executed with `cargo run --example <executable name>`.
+
+Below is a file tree of a typical Burn example package:
+
+```
+examples/burn-example
+├── Cargo.toml
+├── examples
+│   ├── example1.rs
```
Comment (syl20bnr) [inline burn-book/src/examples.md:60]:
```suggestion
│   ├── example1.rs      ---> compiled to example1 binary
```

File: burn-book/src/examples.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+│   ├── example2.rs
```
Comment (syl20bnr) [inline burn-book/src/examples.md:61]:
```suggestion
│   ├── example2.rs      ---> compiled to example2 binary
```

File: burn-book/src/examples.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+│   └── ...
+└── src
+    ├── lib.rs
```
Comment (syl20bnr) [inline burn-book/src/examples.md:64]:
```suggestion
    ├── lib.rs           ---> this is the root file for a library
```

File: burn-book/src/examples.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+| [ONNX Import Inference](https://github.com/tracel-ai/burn/tree/main/examples/onnx-inference)              | Imports an ONNX model pre-trained on MNIST to perform inference on a sample image with Burn.                                                                                                 |
+| [PyTorch Import Inference](https://github.com/tracel-ai/burn/tree/main/examples/pytorch-import)           | Imports a PyTorch model pre-trained on MNIST to perform inference on a sample image with Burn.                                                                                               |
+| [Text Classification](https://github.com/tracel-ai/burn/tree/main/examples/text-classification)           | Trains a text classification transformer model on the AG News or DbPedia dataset. The trained model can then be used to classify a text sample.                                              |
```
Comment (syl20bnr) [inline burn-book/src/examples.md:87]:
```suggestion
| [Text Classification](https://github.com/tracel-ai/burn/tree/main/examples/text-classification)           | Trains a text classification transformer model on the AG News or DbPedia datasets. The trained model can then be used to classify a text sample.                                              |
```

**Review by syl20bnr — APPROVED** [review]
Excellent, I made optional suggestions.

**Comment by laggui** [comment]
> Excellent, I made optional suggestions.

Hah, didn't know we could directly suggest and then commit the suggested changes 🤯 

Suggestions applied! The examples section was your idea after all, so thanks for the feedback 🙏

**Review by nathanielsimard — APPROVED** [review]

**Review by antimora — COMMENTED** [review]
I forgot to submit my review. it was still pending but I had one comment to consider next time.

File: burn-book/src/examples.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+<div class="warning">
+
+Note that some examples use the
+[`datasets` library by HuggingFace](https://huggingface.co/docs/datasets/index) to download the
+datasets required in the examples. This is a Python library, which means that you will need to
+install Python before running these examples. This requirement will be clearly indicated in the
+example's README when applicable.
+
+</div>
```
[the remark this answers is not in the document]
Comment (polarathene) [reply burn-book/src/examples.md:102]: Not likely, unless your docs generator has explicit support for that syntax.

That feature has different names and syntax across multiple popular doc generators (_Admonitions in [Material Mkdocs](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#usage) and [Docusaurus](https://docusaurus.io/docs/markdown-features/admonitions), Asides in [Starlight](https://starlight.astro.build/guides/components/#asides), "Custom Containers" in [VuePress](https://ecosystem.vuejs.press/themes/default/markdown.html#custom-containers)_).

**Pull request merged**